Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
Thanks for explaining.

> I later proposed shipping a separate keg-only homebrew-python-runtime formula that these could use to avoid confusing users who didn't expect the python in $PATH to change. That proposal expired.

I agree with this but just think it's the `python` formula itself that should probably become a keg-only `python@2`.